File: testers/combined/java/generated-and-checked-in/avro_events/combined/avro_events/consumer/DynamicValueHandler.java
package combined.avro_events.consumer;

import com.example.events.DynamicValue;
import combined.avro_events.header.StandardHeaders;
import java.lang.Void;
import java.util.concurrent.CompletableFuture;

/** Handler interface for dynamic-value topic events */
public interface DynamicValueHandler {
  /** Handle a message from the topic */
  CompletableFuture<Void> handle(
    String key,
    DynamicValue value,
    StandardHeaders headers
  );
}

File: testers/combined/kotlin/generated-and-checked-in/avro_events/combined/avro_events/consumer/DynamicValueHandler.kt
package combined.avro_events.consumer

import com.example.events.DynamicValue
import combined.avro_events.header.StandardHeaders
import io.smallrye.mutiny.Uni

/** Handler interface for dynamic-value topic events */
interface DynamicValueHandler {
  /** Handle a message from the topic */
  abstract fun handle(
    key: kotlin.String,
    value: DynamicValue,
    headers: StandardHeaders
  ): Uni<Unit>
}